### PR TITLE
[ONNX] QGemm support

### PR DESCRIPTION
**Pull request by chengven027-intellif**
Hi, TVM
    I try to add a new onnx operator QGEMM. 
    Please help to review,thanks.  @AndrewZhaoLuo

### CODE REVIEW

File: python/tvm/relay/frontend/onnx.py [hunk 1]
```diff
@@ -4898,6 +4898,90 @@ def _impl_v10(cls, inputs, attr, params):
...
+        b = inputs[3]
+        #  a scalar or 1D tensor of size 1 or N
+        b_scale = get_scalar_or_1d_tensor(inputs[4], params)
+        #  a scalar or 1D tensor of size 1 or N
+        b_zp = get_scalar_or_1d_tensor(inputs[5], params, "int32")
+
+        # note that if optional and not provided then value will be None.
+        C = inputs[6] if len(inputs) > 6 else None
+        # must be null or a scalar or 1D tensor of size 1
+        y_scale = inputs[7] if len(inputs) > 7 else None
```
Comment (KJlaccHoeUM9l) [inline python/tvm/relay/frontend/onnx.py:4921]: If an optional input is not passed, then its default value is `None`. So you can simplify this block like this:
```Python
C = inputs[6]
y_scale = inputs[7]
```

File: python/tvm/relay/frontend/onnx.py [hunk 2]
```diff
@@ -4898,6 +4898,90 @@ def _impl_v10(cls, inputs, attr, params):
...
+        # http://www.xavierdupre.fr/app/mlprodict/helpsphinx/onnxops/onnx_commicrosoft_QGemm.html
+
+        a = inputs[0]
+        a_scale = get_scalar(inputs[1], params)
+        a_zp = get_scalar(inputs[2], params, "int32")
+
+        b = inputs[3]
+        #  a scalar or 1D tensor of size 1 or N
+        b_scale = get_scalar_or_1d_tensor(inputs[4], params)
+        #  a scalar or 1D tensor of size 1 or N
```
Comment (KJlaccHoeUM9l) [inline python/tvm/relay/frontend/onnx.py:4915]: Could you please add comments above that describe the shapes for tensors `A` and `B` to understand what `N` is.

File: python/tvm/relay/frontend/onnx.py [hunk 3]
```diff
@@ -4898,6 +4898,90 @@ def _impl_v10(cls, inputs, attr, params):
...
+        assert len(infer_shape(b)) == 2
+        # zero point and scale of input b should have same shape size
+        assert infer_shape(b_scale) == infer_shape(b_zp)
+
+        alpha = float(attr.get("alpha", 1.0))
+        transA = int(attr.get("transA", 0))
+        transB = int(attr.get("transB", 0))
+
+        # get number of channels
+        channels = infer_channels(inputs[3], not transB)
```
Comment (KJlaccHoeUM9l) [inline python/tvm/relay/frontend/onnx.py:4935]: Could you use `b` instead of `inputs[3]` here?

File: python/tvm/relay/frontend/onnx.py (hunk 1 above)
Comment (chengven027-intellif) [reply python/tvm/relay/frontend/onnx.py]: Done.

File: python/tvm/relay/frontend/onnx.py (hunk 3 above)
Comment (chengven027-intellif) [reply python/tvm/relay/frontend/onnx.py]: done,thanks.

File: python/tvm/relay/frontend/onnx.py (hunk 2 above)
Comment (chengven027-intellif) [reply python/tvm/relay/frontend/onnx.py]: yes，tensors `A scale and zero point` don`t  have N， just a scale.And tensor `B scale and zero point`  can have N size 1-D tensor, which means per-column quantization.

**Comment by chengven027-intellif** [comment]
Thanks for your review. I updated the comment. Please review again, Thanks. @KJlaccHoeUM9l

**Comment by AndrewZhaoLuo** [comment]
@tvm-bot rerun

**Comment by AndrewZhaoLuo** [comment]
Sorry, I will take a look tomorrow.

File: python/tvm/relay/frontend/onnx.py [hunk 4]
```diff
@@ -4898,6 +4898,90 @@ def _impl_v10(cls, inputs, attr, params):
         return out
 
 
+class QGemm(OnnxOpConverter):
+    """Operator converter for QGemm."""
+
+    @classmethod
+    def _impl_v1(cls, inputs, attr, params):
+        # http://www.xavierdupre.fr/app/mlprodict/helpsphinx/onnxops/onnx_commicrosoft_QGemm.html
```
Comment (KJlaccHoeUM9l) [inline python/tvm/relay/frontend/onnx.py:4906]: If you want to add a link to the documentation, it's better to point to the official documentation for ORT:
https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#com.microsoft.QGemm

**Review by KJlaccHoeUM9l — APPROVED** [review]

**Review by AndrewZhaoLuo — COMMENTED** [review]
Please rebase to fix CI issues and point documentation to official repo.

Otherwise LGTM.

**Comment by chengven027-intellif** [comment]
Done, Thanks,please review again. @AndrewZhaoLuo

**Review by AndrewZhaoLuo — APPROVED** [review]

**Pull request merged**